### PR TITLE
Add data store

### PR DESCRIPTION
**Pull request by whikernel**
The branch brings a data store : 

 - Upload any files to virtual datastore to share files
 - Password protection as an option 
 - Import in IOC and / or Evidence as an option
 - Reorganise, update all files in datastore
 - Download and direct reference in notes with a prebuilt link
 - Directly paste an image in notes and summary - uses the datastore in bg 
 - UI improvement in notes and summary 

Fixes long waiting https://github.com/dfir-iris/iris-web/issues/30

### CODE REVIEW

File: source/app/static/assets/js/iris/case.summary.js [hunk 1]
```diff
@@ -247,6 +248,98 @@ function auto_remove_typing() {
...
+            editor.insertSnippet('#### ${1:$SELECTION}');
+        }
+    });
+    $('#editor_summary').on('paste', (event) => {
+        event.preventDefault();
+        handle_ed_paste(event);
+    });
+
+    var textarea = $('#case_summary');
+    editor.getSession().on("change", function () {
```
Comment (Ektoplasma) [inline source/app/static/assets/js/iris/case.summary.js:328]: Copy/Paste of pictures are not synchronized by the websocket

File: docker/nginx/nginx.conf [hunk 2]
```diff
@@ -128,7 +128,16 @@ http {
...
+                client_max_body_size        100G;
+                proxy_pass  http://${APP_HOST}:${APP_PORT};
+            }
+
+            location ~ ^/(datastore/file/add|datastore/file/add-interactive) {
+                keepalive_timeout           10m;
+                client_body_timeout         10m;
+                send_timeout                10m;
+                proxy_read_timeout          10m;
+                client_max_body_size        100G;
```
Comment (Ektoplasma) [inline docker/nginx/nginx.conf:140]: From 1G and upper, we still get a 413 from the proxy.

**Review by Ektoplasma — COMMENTED** [review]
On hold. See comments.

File: source/app/static/assets/js/iris/case.summary.js (hunk 1 above)
Comment (whikernel) [reply source/app/static/assets/js/iris/case.summary.js:344]: Fixed in https://github.com/dfir-iris/iris-web/pull/115/commits/14679e4ba3ad736ba8cabf81c72f49f9b5936c0f 
Added event emulation to force sync

File: docker/nginx/nginx.conf (hunk 2 above)
Comment (whikernel) [reply docker/nginx/nginx.conf]: Fixed in https://github.com/dfir-iris/iris-web/pull/115/commits/aef41eb44c65bc082e2defafb6e527b628a66432.  
There was a limit set in Flask itself.

**Review by Ektoplasma — APPROVED** [review]

**Pull request merged**